src/test/java/com/session/service/ClientTest.java

```diff
@@ -67,14 +70,42 @@ public void createSession_emptyEmailAddress_shouldReturnError() {
     }
 
     @Test
-    public void getSessionById_shouldReturnExpectedSession() throws Exception {
-        SessionCreated sessionCreated = client.createNewSession(EMAIL);
-        assertThat(sessionCreated, is(notNullValue()));
-        assertThat(sessionCreated.getId(), not(isEmptyString()));
+    public void getSessionById_shouldReturnExpectedSession() throws IOException {
+        Mockito.when(http.get(
+                eq(HOST),
+                eq("/sessions/" + SESSION_ID),
+                ArgumentMatchers.<ResponseHandler<Session>>any()
+        )).thenReturn(zebedeeSession);
 
-        Session session = client.getSessionByID(sessionCreated.getId());
+        Session session = client.getSessionByID(SESSION_ID);
         assertThat(session, is(notNullValue()));
-        assertThat(session.getId(), equalTo(sessionCreated.getId()));
+        assertThat(session.getId(), equalTo(SESSION_ID));
+    }
+
+    @Test
+    public void getSessionById_whenNullSessionReturned_shouldReturnError() throws IOException {
+        Mockito.when(http.get(
+                eq(HOST),
+                eq("/sessions/" + SESSION_ID),
+                ArgumentMatchers.<ResponseHandler<Session>>any()
+        )).thenReturn(null);
+
+        SessionClientException sessionClientException = assertThrows(SessionClientException.class, () ->
+                client.getSessionByID(SESSION_ID));
+        assertThat(sessionClientException.getMessage(), is("invalid session"));
+    }
+
+    @Test
+    public void getSessionById_whenUnableToConnect_shouldReturnError() throws IOException {
```

Comment:
The test it's self is perfectly fine. The naming however suggests possibly too much knowledge of the `http`  dependencies internal workings. I think this test case should be something like `getSessionById_httpThrowsIOException_shouldReturnError`

Remember the `Http` object is an external dependency of this class so from this class's POV its knowledge of what `Http` can do should be strictly limited to public API details - if you keep this in mind you'll avoid tightly coupling the objects together and your tests will be simpler.

Reply:
That sounds much better.